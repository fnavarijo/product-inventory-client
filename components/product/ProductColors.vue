<template>
  <div class="my-4 flex items-center">
    <span class="mr-4">Colores</span>
    <div
      v-for="({ color, name }, index) in colors"
      :key="index"
      class="rounded-full h-6 w-6"
      :style="{ backgroundColor: color }"
      :title="name"
    />
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

import { Color } from '@/api/types';

export default Vue.extend({
  name: 'ProductColors',
  props: {
    colors: {
      type: Array as PropType<Array<Color>>,
      default: () => [],
    },
  },
});
</script>
