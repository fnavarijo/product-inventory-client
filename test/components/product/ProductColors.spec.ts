import { mount } from '@vue/test-utils';

import ProductColors from '@/components/product/ProductColors.vue';

describe('ProductColors component', () => {
  it('should render component with correct styling', () => {
    const wrapper = mount(ProductColors, {
      propsData: {
        colors: [
          { name: 'Black', color: '#000000' },
          { name: 'White', color: '#ffffff' },
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
