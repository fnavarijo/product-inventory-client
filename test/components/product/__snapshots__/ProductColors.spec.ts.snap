// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductColors component should render component with correct styling 1`] = `
<div class="my-4 flex items-center"><span class="mr-4">Colores</span>
  <div title="Black" class="rounded-full h-6 w-6" style="background-color: rgb(0, 0, 0);"></div>
  <div title="White" class="rounded-full h-6 w-6" style="background-color: rgb(255, 255, 255);"></div>
</div>
`;
